numpy/_core/src/umath/ufunc_type_resolution.c

```diff
@@ -1115,12 +1115,46 @@ PyUFunc_MultiplicationTypeResolver(PyUFuncObject *ufunc,
     type_num2 = PyArray_DESCR(operands[1])->type_num;
 
     /* Use the default when datetime and timedelta are not involved */
-    if (!PyTypeNum_ISDATETIME(type_num1) && !PyTypeNum_ISDATETIME(type_num2)) {
+    if (!PyTypeNum_ISDATETIME(type_num1) && !PyTypeNum_ISDATETIME(type_num2)
+        && !((PyTypeNum_ISSTRING(type_num1) && PyTypeNum_ISINTEGER(type_num2))
+             || (PyTypeNum_ISINTEGER(type_num1) && PyTypeNum_ISSTRING(type_num2)))) {
         return PyUFunc_SimpleUniformOperationTypeResolver(ufunc, casting,
                     operands, type_tup, out_dtypes);
     }
 
-    if (type_num1 == NPY_TIMEDELTA) {
+    if (PyTypeNum_ISSTRING(type_num1) || PyTypeNum_ISSTRING(type_num2)) {
```

Comment:
argh, this is not something you need to deal with here but we should probably deprecate this macro (if possible) and replace it with ISLEGACYSTRING or similar. Note to other reviewers, this is only true for bytes and unicode.

Reply:
True.